lib/Model/Listing/Dao/QueryBuilderHelperTrait.php

```diff
@@ -23,13 +23,26 @@
 trait QueryBuilderHelperTrait
 {
     /**
-     * @var callable|null
+     * @var callable[]
      */
-    protected $onCreateQueryBuilderCallback;
+    protected array $queryBuilderProcessors = [];
 
     public function onCreateQueryBuilder(?callable $callback): void
     {
-        $this->onCreateQueryBuilderCallback = $callback;
+        $this->discardQueryBuilderProcessors();
+        if (is_callable($callback)) {
+            $this->addQueryBuilderProcessor($callback);
+        }
```

Comment:
Reverting 
`protected $onCreateQueryBuilderCallback;`
and
```suggestion
        $this->onCreateQueryBuilderCallback = $callback;
        $this->queryBuilderProcessors = [$callback];
```
```
        foreach ($this->queryBuilderProcessors as $processor) {
            if (is_callable($callback)) {
                $processor($queryBuilder);
            }
        }
```

should be fine, right?

Reply:
As new property is used exclusively within the trait - I don't see any BC breaking. New property is used only in the trait, and public API is exactly the same. Also method's result is the same.

Reply:
As a trait is like a code snippet that is copied into a class, *everything* in it is "public API", even `private` methods or properties, because classes that use the trait can access them.

But as the property is `protected`, even *child classes* of the ones that use the trait can access it.

So yes, this is definitely a BC break.